core/src/components/segment/segment.tsx

```diff
@@ -431,6 +439,76 @@ export class Segment implements ComponentInterface {
     this.checked = current;
   }
 
+  private getSegmentButton = (selector: 'first' | 'last' | 'next' | 'previous' | 'focused'): HTMLIonSegmentButtonElement | null => {
+    const buttons = this.getButtons().filter(button => !button.disabled);
+    const currIndex = buttons.findIndex(button =>
+      (this.keyboardBehavior === 'select' && this.checked)
+      ? button === this.checked
+      : button.classList.contains('ion-focused'
+    ));
```

Comment:
Actually thought of a better solution:

```suggestion
    const currIndex = buttons.findIndex(button => button === document.activeElement || button === this.checked);
```

This returns the index if the button we are looking at is either the focused element or the checked/selected element. This also avoids iterating through all the classes on each button.

Reply:
We need to keep the logic for checking whether `selectOnFocus` is true or false, but I changed the `classList.contains` to `document.activeElement` as per your suggestion. 

Also, I changed it to an `if` statement and added a comment for readability.

Reply:
But if `selectOnFocus: true`, isn't the current button focused too?

Reply:
Ah! I was seeing weird behavior when trying your suggestion, but it was fixed by just checking for `button === document.activeElement` rather than `button === document.activeElement || button === this.checked`. 

The checked button is only relevant if it is also the `activeElement`, so checking for it was a redundant holdover from my initial commit. 

Thanks!